### DISCUSSION TITLE
Simple authorization of authenticated users by linux groups or roles in a file?

### DISCUSSION THREAD

**Discussion by chrisinmtown** [OpenIDC/mod_auth_openidc#709] [Q&A] [answered]
I need to restrict which openidc-authenticated users can access a web site. I read about mod-auth-openidc authorization here: https://github.com/zmartzone/mod_auth_openidc/wiki/Authorization

Please tell me, is there a basic option to restrict users based on local resources such as flat files or linux groups? Maybe a separate Apache HTTPD module that will mesh well with open id connect?

I see from the documentation that I can restrict authenticated users based on claims provided by the provider or from an LDAP server via mod_authnz_ldap. My internal provider doesn't support authorization claims like this, just authentication of everyone in the company. If there is no simple option, maybe I can stand up a LDAP server; do you have a recommendation?

Thanks in advance.

**Comment by zandbelt** [comment, marked as answer]
Whatever you'd put into that file, you can put directly in the Apache config.

**Reply by chrisinmtown** [reply]
Thanks, I think I see your point -- I can just list the authorized user IDs in the httpd-ssl.conf file?

**Reply by chrisinmtown** [reply]
I tried to be clear in my question, I do not want to change the Apache HTTPD configuration directly for each authorization change, but I guess that's the only simple file-based approach supported today.

**Reply by zandbelt** [reply]
one can use Include https://httpd.apache.org/docs/2.4/mod/core.html#include

**Reply by chrisinmtown** [reply]
Thanks for mentioning the Apache HTTPD Include directive but that's still functionally the Apache HTTPD config in my opinion, and I believe a change to an Include-d file still requires restart of HTTPD.

**Reply by hmoffatt** [reply]
All or most of those auth directives can go in a .htaccess file instead which doesn't require a server restart.

**Reply by chrisinmtown** [reply]
Would you please show an example of this?

**Reply by hmoffatt** [reply]
I didn't try, but based on https://github.com/zmartzone/mod_auth_openidc/wiki/Authorization#2-mod_authnz_ldap I think you should be able to write just `Require user <username>` where username is whatever the username is from your IDP.

You can put that in .htaccess just the same as in a <Location> or <Directory> clause in your apache config.

If that works then I think you should be able to build a group file listing the users (since your IDP doesn't provide any group information?) and use that. See https://httpd.apache.org/docs/2.4/mod/mod_authz_groupfile.html
